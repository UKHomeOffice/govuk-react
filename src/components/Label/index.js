// https://github.com/alphagov/govuk-frontend/tree/master/src/components/

import glamorous from "glamorous";

import { ERROR_COLOUR } from "govuk-colours";
import { MEDIA_QUERIES, SITE_WIDTH, SPACING } from "../../constants/index";

const Label = glamorous.label(
  {
    display: "flex",
    flexDirection: "column",
    boxSizing: "border-box",
    width: "100%",
    ":after": {
      content: "''",
      display: "table",
      clear: "both"
    },
    [MEDIA_QUERIES.LARGESCREEN]: {
      maxWidth: SITE_WIDTH
    }
  },
  ({ error }) => ({
    borderLeft: error ? `4px solid ${ERROR_COLOUR}` : "",
    marginRight: error ? SPACING.SCALE_3 : 0,
    paddingLeft: error ? SPACING.SCALE_2 : 0
  })
);

export default Label;
