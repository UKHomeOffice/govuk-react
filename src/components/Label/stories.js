import React from 'react';
import { storiesOf } from '@storybook/react';
import Label from '.';

storiesOf('Label', module).add('Component default', () => (
  <Label>Example</Label>
));
