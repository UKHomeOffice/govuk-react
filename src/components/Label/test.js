import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, render } from "enzyme";
import Label from "./";

describe("Label", () => {
  const example = "example";
  const wrapper = <Label>{example}</Label>;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it("matches snapshot", () => {
    expect(shallow(wrapper)).toMatchSnapshot(`enzyme.shallow`);
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
    expect(render(wrapper)).toMatchSnapshot(`enzyme.render`);
  });
});
