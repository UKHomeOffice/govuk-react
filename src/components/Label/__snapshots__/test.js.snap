// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label matches snapshot: enzyme.mount 1`] = `
<glamorous(label)>
  <label
    className="css-16y5nda"
  >
    example
  </label>
</glamorous(label)>
`;

exports[`Label matches snapshot: enzyme.render 1`] = `
<label
  class="css-16y5nda"
>
  example
</label>
`;

exports[`Label matches snapshot: enzyme.shallow 1`] = `
<label
  className="css-16y5nda"
>
  example
</label>
`;
